.teamcity/settings.kts

```diff
@@ -74,4 +78,171 @@ object Chocolatey : BuildType({
+        //         content = """
+        //             docker run --rm chocolatey/choco:latest-windows choco.exe --version
+        //             exit ${'$'}LastExitCode
+        //         """.trimIndent()
+        //     }
+        // }
+    }
+
+    triggers {
+        finishBuildTrigger {
+            buildType = "Chocolatey"
+            successfulOnly = true
+            branchFilter = """
+                +:tags/*
+            """.trimIndent()
+        }
+    }
+
+    dependencies {
+        dependency(AbsoluteId("Chocolatey")) {
+            snapshot {
+                onDependencyFailure = FailureAction.FAIL_TO_START
+                synchronizeRevisions = false
+            }
+
+            artifacts {
+                artifactRules = "chocolatey.%env.CHOCOLATEY_VERSION%.nupkg=>%system.teamcity.build.checkoutDir%\\code_drop\\Packages\\Chocolatey"
+            }
+        }
+    }
+
+    requirements {
+        contains("docker.server.osType", "windows")
+        exists("docker.server.version")
+    }
+})
+
+object ChocolateyPosix : BuildType({
+    id = AbsoluteId("ChocolateyPosix")
+    name = "Docker (Linux)"
+
+    params {
+        param("env.CHOCOLATEY_VERSION", "%dep.Chocolatey.build.number%")
+        param("env.CHOCOLATEY_OFFICIAL_KEY", "%system.teamcity.build.checkoutDir%/chocolatey.official.snk")
+        password("env.GITHUB_PAT", "%system.GitHubPAT%", display = ParameterDisplay.HIDDEN, readOnly = true)
+        param("env.vcsroot.branch", "%vcsroot.branch%")
+        param("env.Git_Branch", "%teamcity.build.vcs.branch.Chocolatey_ChocolateyVcsRoot%")
+        param("teamcity.git.fetchAllHeads", "true")
+    }
+
+    vcs {
+        root(DslContext.settingsRoot)
+    }
+
+    artifactRules = """
+        code_drop/Packages/Chocolatey/chocolatey.*.tar.gz
+    """.trimIndent()
+
+    steps {
+        step {
+            name = "Load Key"
+            conditions {
+                doesNotContain("teamcity.serverUrl", "-dev")
+            }
+            type = "StrongNameKeyLinux"
+        }
+
+        script {
+            name = "Install Prerequisites"
+            scriptContent = """
+                sudo apt update
+                sudo apt install dirmngr gnupg apt-transport-https ca-certificates software-properties-common
+                sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 3FA7E0328081BFF6A14DA29AA6A19B38D3D831EF
+                sudo apt-add-repository 'deb https://download.mono-project.com/repo/ubuntu stable-focal main'
+                sudo apt install mono-complete -y
+            """.trimIndent()
+        }
+
+        script {
+            name = "Build Chocolatey"
+            scriptContent = """
+                ./build.official.sh --verbosity=diagnostic --nuget_useinprocessclient=false
+            """.trimIndent()
+        }
+
+        // Please note that this method will need to be changed to some form of CD after we lock agents down
+        script {
```

Comment:
Have we added a task somewhere to capture this? And by that I mean, using something other that curl to upload the artifact to the release.

Reply:
Sounds good.